geth/account/adapter_test.go

```diff
@@ -0,0 +1,37 @@
+package account
```

Comment:
Why is this a white-box test and not `account_test`?

Reply:
@themue Why do you against white-box testing? We have e2e black-box tests which test  expectation of the external behaviour.

Reply:
Mostly it's to separate implementation detail from testing code. IMHO the testing code for a type/function should have the same knowledge about it as the user of that type, so only public interfaces. And the tests only assert the expectations they have for a type/function.

E2E is something different, integrating different layers. Unit tests stub those layers if needed to only focus on the tested component. Here the stub ensures the needed response for given arguments to have deterministic tests.